Source/Core/UICommon/UICommon.cpp

```diff
@@ -157,6 +157,8 @@ void CreateDirectories()
   File::CreateFullPath(File::GetUserPath(D_USER_IDX));
   File::CreateFullPath(File::GetUserPath(D_CACHE_IDX));
   File::CreateFullPath(File::GetUserPath(D_CONFIG_IDX));
+  File::CreateFullPath(File::GetUserPath(D_DUMPFRAMES_IDX));
```

Comment:
This should be reverted and the other CreateFullPath calls for dump paths should be moved to UICommon::CreateDumpPath.

Reply:
Moving
`File::CreateFullPath(File::GetUserPath(D_DUMPFRAMES_IDX));
File::CreateFullPath(File::GetUserPath(D_DUMPAUDIO_IDX));
File::CreateFullPath(File::GetUserPath(D_DUMPOBJECTS_IDX));`
to UICommon::CreateDumpPath doesn't create the folders but UICommon::CreateDirectories() does. Am I doing something wrong or is this a bug?

Reply:
Oh yeah, I think there's a bug. Can you change these lines:

https://github.com/dolphin-emu/dolphin/blob/21898a0c5a2bf55d0d48fde10604435d7336b07e/Source/Core/UICommon/UICommon.cpp#L53-L55

to:

```c++
  if (!path.empty())
    File::SetUserPath(D_DUMP_IDX, path + '/');
```

Otherwise dump dirs won't get created if the user has not configured a custom dump path.